cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1207,27 +1207,43 @@ impl FromStr for CatalogPackage {
     ///
     ///     attribute_path ::= <install_id> | <attribute_path_rest>.<install_id>
     ///     attribute_path_rest ::= <identifier> | <attribute_path_rest>.<identifier>
-    ///     install_id ::= <identifier>
+    ///     install_id ::= <identifier> | @<identifier>
     ///
     ///     version ::= <string> # interpreted as semver or plain version by the resolver
     /// ```
-    /// Todo: this does currently _not_ handle any more pathological cases like
-    ///  - `@` in the version string (the last `@` is the delimiter)
     fn from_str(descriptor: &str) -> Result<Self, ManifestError> {
-        let (attr_path, version) = match descriptor.split_once('@') {
-            Some((attr_path, version)) if !version.is_empty() => {
-                (attr_path.to_string(), Some(version.to_string()))
-            },
-            Some(_) => {
+        fn split_version(haystack: &str) -> (usize, Option<&str>) {
+            let Some(version_at) = haystack.find('@') else {
+                return (haystack.len(), None);
+            };
+
+            if version_at == 0 {
```

Comment:
```suggestion
            // If the descriptor begins with "@", treat the entire thing as an attr_path
            // TODO: this doesn't support versions for attributes that begin with "@"
            if version_at == 0 {
```

Reply:
iterative solution also supports versions where the attrpath starts with @ now